views/hud.html

```diff
@@ -0,0 +1,132 @@
+	<link rel="apple-touch-icon" sizes="120x120" href="/images/apple-touch-icon-120x120.png">
+	<link rel="apple-touch-icon" sizes="144x144" href="/images/apple-touch-icon-144x144.png">
+	<link rel="apple-touch-icon" sizes="152x152" href="/images/apple-touch-icon-152x152.png">
+	<link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-180x180.png">
+  
+	<style type="text/css">
+		@import url("//fonts.googleapis.com/css?family=Open+Sans:700");
+    	body {
+			text-align: center;
+			margin: 0 0;
+			padding: 0;
+			overflow: hidden;
+			font-family: 'Open Sans';
+			color: white;
+			background-color: black;
+			-moz-transform: scale(1, -1);
+			-webkit-transform: scale(1, -1);
+			-o-transform: scale(1, -1);
+			-ms-transform: scale(1, -1);
+			transform: scale(1, -1);
+		}
+		main {
+			display: -webkit-box;
+			display: -ms-flexbox;
+			display: -webkit-flex;
+			display: flex;
+			-webkit-box-align: center;
+			-ms-flex-align: center;
+			-webkit-align-items: center;
+			align-items: center;
+			height: 100vh;
+		}
+		.inner {
+			width: 100%;
+			-webkit-transform: translateY(-2%);
+		}
+		#trend {
+			display: -ms-flexbox;
+			display: -webkit-flex;
+			display: flex;
+			-ms-flex-align: center;
+			-webkit-align-items: center;
+			align-items: center;
+			justify-content: center;
+			-webkit-flex-direction: row;
+			flex-direction: row;
+		}
+		#bgnow, #arrowDiv {
+			display: flex;
+			flex-grow: 0;
+			font-weight: 700;
+			font-size: 30vmin;
+			padding: 0 20px;
+			margin: 0;
+		}
+		img#arrow {
+			height: 15vmin;
+		}font-size: 15vmin;
+		}
+		.stale {
+			text-decoration: line-through;
+		}
+  </style>
+</head>
+
+<body>
+
+	<main>
+		<div class="inner">
+			<div id="trend">
+				<div id="bgnow"></div>
+				<div id="arrowDiv"><img id="arrow" src=""/></div>
+			</div>
+		</div>
+	</main>
+	
+	<script src="/api/v1/status.js"></script>
+	<script src="/js/bundle.js?v=2c1zd4vahmonsjvk"></script>
+
+	<script type="text/javascript">
+		function query ( ) {
+			console.log('query');
+			$.ajax('/api/v1/entries/current.json', { success: render });
+		}
+		function render (xhr) {
+			console.log('xhr', xhr);
+			
+			var rec = xhr[0];
+			var last = new Date(rec.date);
+			var now = new Date( );
+			
+			// Convert BG to mmol/L if necessary.
+			if (window.serverSettings.settings.units == 'mmol') {
+				rec.sgv = Nightscout.units.mgdlToMMOL(rec.sgv);
+			}
+			
+			// Insert the BG value text.
+			$('#bgnow').html(rec.sgv);
+			
+			// Insert the trend arrow.
+			$('#arrow').attr('src', 'images/'+rec.direction+'.svg');
+			
+			// Time before data considered stale.
+			var staleMinutes = 13;										
+			var threshold = 1000 * 60 * staleMinutes;
+			
+			// Toggle stale if necessary.
+			$('#bgnow').toggleClass('stale', (now - last > threshold));
+		}
+		function init ( ) {
+			console.log('init');
+			query( );
+			setInterval(query, 1 * 60 * 1000);
```

Comment:
see https://github.com/nightscout/cgm-remote-monitor/blob/7c272ec67a7795e0f4edf7ff9abdfac1bd1308a0/lib/client/index.js on how to use `socket.io` instead of a busy wait.

Reply:
I think this is out of my skillset unfortunately... :/ any pointers @PieterGit ?

Reply:
@haspden I already gave you a pointer 😄. You can always extend your skill set by asking for help. To be more specific have a look at this part of the client code https://github.com/nightscout/cgm-remote-monitor/blob/dev/lib/client/index.js#L897 . You need to set up a socket, and implement a custom `dataUpdate` method. 

Perhaps we need to make a minimal client-handler javascript so that that can be re-used/extended by the main Nightscout screens, and the `clock.html` and `hud.html` views. That would also solve the problems with token-based authentication. Since this method is also approved for `clock.html` we might postpone this part to 0.12 release.

I really hope @sulkaharo or @jasoncalabrese can join this discussion

Reply:
Since this is essentially just a clone of the three existing (and stable) clock views, I think it’s reasonable to implement it as-is for 0.11 and then work on re-architecting all of the views (if that’s agreed to be appropriate) for 12.